### DISCUSSION TITLE
Help with SSL connection on Amazon DocumentDB

### DISCUSSION THREAD

**Discussion by upcase-chefs** [payloadcms/payload#52] [Q&A] [answered]
Hey folks! First, congratulations. Payload CMS looks amazing. 
We, here in Chefsclub, are excited to use Payload in some of our usecases.
A little (and perhaps silly) question comes up to my mind:
We are trying to connect payload on a DocumentDB cluster that requires a SSL .pem certificate. Since we can just pass on the init the URI, we are not getting any success to connect without (1 have to disable ssl :(  or 2 use it on the same VPC). 
Is there any plans to make the parameter passed on the Payload init to include the mongoose options, or at least the CA SSL certificate?

Thank you so much.
Filipe Baeta 
CTO @Chefsclub

**Comment by jmikrut** [comment]
Hi @upcase-chefs — great question and great request. We would be happy to support the passing of Mongoose options.

Give us a bit here and we will make it happen! You can expect this functionality to be released in 1 to 2 days from now.

We will report back to this discussion when we have it added!

**Reply by upcase-chefs** [reply]
Hey Folks, 

My suggested steps: 

1. on the index.js add the follow parameter besides the url: mongoAdditionalOptions

On the connect.js, receive the mongoAdditionalOptions object and append it to the connectionOptions...

var finalOptions = {...connectionOptions, ...mongoAdditionalOptions} 
Also, on the index.js it's needed read Amazon's .pem file and pass the buffer (or the string. I'm not quite sure) to the mongoose parameters.

https://mongoosejs.com/docs/tutorials/ssl.html

I hope this helps. 

We are Eager to buy the Payload licenses. But we still need to fix this one and another bug on Relationship that I just opened there. This is blocking our workload to be implemented.

Cheers.
Chefsclub team.

**Reply by DanRibbens** [reply]
This is similar to what I came up with.

I named it `mongoOptions` instead of `mongoAdditionalOptions`, but the functionality is the same as you described.

Here is the PR:
https://github.com/payloadcms/payload/pull/56/files

**Comment by DanRibbens** [comment, marked as answer]
This feature is up now in the published v0.2.10! You can update your package.json Payload dependency to `"^0.2.10"` get this change.

**Reply by upcase-chefs** [reply]
Hey Dan! Thank you and the team so much! 
Now the fun begins :)

**Reply by prove-ability** [reply]
```yarn update payload@0.2.10``` did i misunderstand?

<img width="973" alt="스크린샷 2022-06-08 15 32 13" src="https://user-images.githubusercontent.com/59461181/172547763-ca300ca2-9677-4e79-8ee1-b96f25a398e5.png">

Thanks in advance!

**Reply by jmikrut** [reply]
Hey @prove-ability — Payload version `0.2.10` is now quite out of date. This discussion was from Feb 2021. We're now at `0.17.3`. You should use the latest version of Payload instead.
